src/test/java/maksym/fedorenko/bookstore/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,267 @@
+import maksym.fedorenko.bookstore.repository.BookRepository;
+import maksym.fedorenko.bookstore.repository.CartItemRepository;
+import maksym.fedorenko.bookstore.repository.ShoppingCartRepository;
+import maksym.fedorenko.bookstore.repository.UserRepository;
+import maksym.fedorenko.bookstore.service.impl.ShoppingCartServiceImpl;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.Authentication;
+
+@ExtendWith(MockitoExtension.class)
+public class ShoppingCartServiceTest {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private CartItemRepository cartItemRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Spy
+    private CartItemMapper cartItemMapper = new CartItemMapperImpl();
+    @Spy
+    private ShoppingCartMapper shoppingCartMapper = new ShoppingCartMapperImpl(cartItemMapper);
+    @Mock
+    private Authentication authentication;
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @Test
+    @DisplayName("Get user cart with valid authentication")
+    void getUserCart_ValidAuthentication_ShouldReturnCartDto() {
+        String username = "user@gmail.com";
+        ShoppingCart cart = new ShoppingCart();
+        cart.setId(1L);
+
+        when(authentication.getName()).thenReturn(username);
+        when(shoppingCartRepository.findByUserEmail(username)).thenReturn(Optional.of(cart));
+
+        CartDto actual = shoppingCartService.getUserCart(authentication);
+        assertThat(actual)
+                .hasFieldOrPropertyWithValue("id", 1L)
+                .extracting(CartDto::cartItems)
+                .asInstanceOf(InstanceOfAssertFactories.list(CartItemDto.class))
+                .hasSize(0);
+    }
+
+    @Test
+    @DisplayName("Get user cart with invalid authentication")
+    void getUserCart_InvalidAuthentication_ShouldThrowException() {
+        String username = "user@gmail.com";
+
+        when(authentication.getName()).thenReturn(username);
+        when(shoppingCartRepository.findByUserEmail(username)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> shoppingCartService.getUserCart(authentication))
+                .isInstanceOf(EntityNotFoundException.class)
+                .hasMessage("Can't find cart for user with email: %s".formatted(username));
+    }
+
+    @Test
+    @DisplayName("Add valid item to cart")
+    void addCartItem_ValidRequestDto_ShouldReturnCartDto() {
+        final String username = "user@gmail.com";
+        final CreateCartItemRequestDto requestDto = new CreateCartItemRequestDto(1L, 2);
```

Comment:
```suggestion
        String username = "user@gmail.com";
        CreateCartItemRequestDto requestDto = new CreateCartItemRequestDto(1L, 2);
```

**Applicable to all places**

Reply:
checkstyle swear that the ad is far from the first use